Adyen/client.py

```diff
@@ -116,70 +117,65 @@ def __init__(
-        """This returns the Adyen API endpoint based on the provided platform,
-        service and endpoint.
-
-        Args:
-            platform (str): Adyen platform, ie 'live' or 'test'.
-            service (str): API service to place request through.
-            endpoint (str): the API endpoint to call.
-        """
-        if platform == "live" and self.live_endpoint_prefix:
-            base_uri = settings.PAL_LIVE_ENDPOINT_URL_TEMPLATE.format(
-                self.live_endpoint_prefix
-            )
-        else:
-            base_uri = settings.BASE_PAL_URL.format(platform)
-
-        if service == "Recurring":
-            api_version = self.api_recurring_version
-        elif service == "Payout":
-            api_version = self.api_payout_version
-        elif service == "BinLookup":
-            api_version = self.api_bin_lookup_version
-        elif service == "terminal":
-            base_uri = settings.BASE_TERMINAL_URL.format(platform)
-            api_version = self.api_terminal_version
-        else:
-            api_version = self.api_payment_version
-        return '/'.join([base_uri, service, api_version, endpoint])
-
-    def _determine_checkout_url(self, platform, endpoint):
-        """This returns the Adyen API endpoint based on the provided platform,
-        service and endpoint.
-
-        Args:
-            platform (str): Adyen platform, ie 'live' or 'test'.
-            endpoint (str): The API endpoint to call.
-        """
-        api_version = self.api_checkout_version
-        if platform == "test":
-            base_uri = settings.ENDPOINT_CHECKOUT_TEST
-        elif self.live_endpoint_prefix is not None and platform == "live":
-            base_uri = settings.ENDPOINT_CHECKOUT_LIVE_SUFFIX.format(
-                self.live_endpoint_prefix)
-        elif self.live_endpoint_prefix is None and platform == "live":
-            errorstring = """Please set your live suffix. You can set it
-                   by running 'settings.
-                   ENDPOINT_CHECKOUT_LIVE_SUFFIX = 'Your live suffix'"""
-            raise AdyenEndpointInvalidFormat(errorstring)
-        else:
-            raise AdyenEndpointInvalidFormat("invalid config")
-
-        if endpoint == "originKeys":
-            api_version = self.api_checkout_utility_version
-
-        return '/'.join([base_uri, api_version, endpoint])
-
-    def _determine_management_url(self, platform, endpoint):
-        api_version = self.api_management_version
-        if platform == "test":
-            base_uri = settings.ENDPOINT_MANAGEMENT_TEST
-        else:
-            base_uri = settings.ENDPOINT_MANAGEMENT_LIVE
+    def _determine_base_uri_and_version(self, platform, service):
+        versions_and_urls = {
+            'Recurring': {
+                'version': self.api_recurring_version,
+                'base_url': {
+                    'live': settings.PAL_LIVE_ENDPOINT_URL_TEMPLATE.format(self.live_endpoint_prefix) + '/Recurring',
```

Comment:
We can reuse this url: 

```
live_pal_url = settings.PAL_LIVE_ENDPOINT_URL_TEMPLATE.format(self.live_endpoint_prefix)
```

```suggestion
                    'live': live_pal_url + '/Recurring',
```